cpp-ch/local-engine/Shuffle/ShuffleWriter.cpp

```diff
@@ -25,13 +25,14 @@ using namespace DB;
 namespace local_engine
 {
 ShuffleWriter::ShuffleWriter(
-    jobject output_stream, jbyteArray buffer, const std::string & codecStr, bool enable_compression, size_t customize_buffer_size)
+    jobject output_stream, jbyteArray buffer, const std::string & codecStr, jint level, bool enable_compression, size_t customize_buffer_size)
 {
     compression_enable = enable_compression;
     write_buffer = std::make_unique<WriteBufferFromJavaOutputStream>(output_stream, buffer, customize_buffer_size);
     if (compression_enable)
     {
-        auto codec = DB::CompressionCodecFactory::instance().get(boost::to_upper_copy(codecStr), {});
+        auto family = boost::to_upper_copy(codecStr);
+        auto codec = DB::CompressionCodecFactory::instance().get(family, family == "ZSTD" ? std::optional<int>(level) : std::nullopt);
```

Comment:
Is level only useful for ZSTD CODEC ?  

Maybe we can use `xxx` instead of `std::optional<int>(xxx)`,

Reply:
the default value of level is INT_MIN, it is ok to pass it directly to c++ codes ?